core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -727,6 +798,7 @@ object SnappyStoreHiveCatalog {
 
   private[this] var relationDestroyVersion = 0
   private val relationDestroyLock = new ReentrantReadWriteLock()
+  private val alterTableLock = new Object
```

Comment:
Purpose of this lock? The relationDestroyLock was for a different reason, namely to protect the relationDestroyVersion so that any drops from other catalog instances cause all other catalog instances to also discard their cached value in cachedDataSourceTables.